daemon/cmd/watchdogs.go

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cmd
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+
+	"github.com/cilium/cilium/pkg/controller"
+	"github.com/cilium/cilium/pkg/datapath/loader"
+	"github.com/cilium/cilium/pkg/endpoint"
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/promise"
+)
+
+const epBPFProgWatchdog = "ep-bpf-prog-watchdog"
+
+type epBPFProgWatchdogConfig struct {
+	EndpointBPFProgWatchdogInterval time.Duration
+	EndpointBPFProgWatchdogEnabled  bool
+}
+
+func (c epBPFProgWatchdogConfig) Flags(flags *pflag.FlagSet) {
+	flags.Bool("endpoint-bpf-prog-watchdog-enabled", c.EndpointBPFProgWatchdogEnabled,
+		"Enable endpoint BPF program watchdog")
+
+	flags.Duration("endpoint-bpf-prog-watchdog-rate", c.EndpointBPFProgWatchdogInterval,
```

Comment:
This should be `"endpoint-bpf-prog-watchdog-interval"`. CI currently fails due to the mismatch:
```
2023-09-15T00:27:07.709604887Z level=error msg="Invoke failed" ="cmd.registerEndpointBPFProgWatchdog (watchdogs.go:62)" error="could not build arguments for function \"github.com/cilium/cilium/daemon/cmd\".registerEndpointBPFProgWatchdog (/go/src/github.com/cilium/cilium/daemon/cmd/watchdogs.go:62): failed to build cmd.epBPFProgWatchdogConfig: received non-nil error from function \"github.com/cilium/cilium/pkg/hive/cell\".(*config[...]).Apply.func2 (/go/src/github.com/cilium/cilium/pkg/hive/cell/config.go:136): failed to unmarshal config struct cmd.epBPFProgWatchdogConfig: 2 error(s) decoding:\n\n* '' has invalid keys: endpoint-bpf-prog-watchdog-rate\n* '' has unset fields: EndpointBPFProgWatchdogInterval.\nHint: field 'FooBar' matches flag 'foo-bar', or use tag `mapstructure:\"flag-name\"` to match field with flag" subsys=hive
```

Reply:
To test the agent hive instantiation: `go test ./daemon/cmd -test.run TestAgentCell -test.v`.

Reply:
Actually even better is:
```
% go run ./daemon hive                                                                           1 ↵
level=info msg="Memory available for map entries (0.003% of 67377799168B): 168444497B" subsys=config
level=info msg="option bpf-ct-global-tcp-max set by dynamic sizing to 591033" subsys=config
level=info msg="option bpf-ct-global-any-max set by dynamic sizing to 295516" subsys=config
level=info msg="option bpf-nat-global-max set by dynamic sizing to 591033" subsys=config
level=info msg="option bpf-neigh-global-max set by dynamic sizing to 591033" subsys=config
level=info msg="option bpf-sock-rev-map-max set by dynamic sizing to 295516" subsys=config
level=warning msg="Running Cilium with \"kvstore\"=\"\" requires identity allocation via CRDs. Changing identity-allocation-mode to \"crd\"" subsys=config
level=error msg="Invoke failed" ="cmd.registerEndpointBPFProgWatchdog (watchdogs.go:62)" error="could not build arguments for function \"github.com/cilium/cilium/daemon/cmd\".registerEndpointBPFProgWatchdog (/home/jussi/work/cilium2/daemon/cmd/watchdogs.go:62): failed to build cmd.epBPFProgWatchdogConfig: received non-nil error from function \"github.com/cilium/cilium/pkg/hive/cell\".(*config[...]).Apply.func2 (/home/jussi/work/cilium2/pkg/hive/cell/config.go:136): failed to unmarshal config struct cmd.epBPFProgWatchdogConfig: 2 error(s) decoding:\n\n* '' has invalid keys: endpoint-bpf-prog-watchdog-rate\n* '' has unset fields: EndpointBPFProgWatchdogInterval.\nHint: field 'FooBar' matches flag 'foo-bar', or use tag `mapstructure:\"flag-name\"` to match field with flag" subsys=hive
level=fatal msg="Failed to populate object graph" error="could not build arguments for function \"github.com/cilium/cilium/daemon/cmd\".registerEndpointBPFProgWatchdog (/home/jussi/work/cilium2/daemon/cmd/watchdogs.go:62): failed to build cmd.epBPFProgWatchdogConfig: received non-nil error from function \"github.com/cilium/cilium/pkg/hive/cell\".(*config[...]).Apply.func2 (/home/jussi/work/cilium2/pkg/hive/cell/config.go:136): failed to unmarshal config struct cmd.epBPFProgWatchdogConfig: 2 error(s) decoding:\n\n* '' has invalid keys: endpoint-bpf-prog-watchdog-rate\n* '' has unset fields: EndpointBPFProgWatchdogInterval.\nHint: field 'FooBar' matches flag 'foo-bar', or use tag `mapstructure:\"flag-name\"` to match field with flag" subsys=hive
```

Reply:
Nice! I did not know that. Thank you for the pointer!